src/DataSource/DB.hs

```diff
@@ -396,9 +410,9 @@ createSchema conn = liftIO $ execute_ conn
 \       FOREIGN KEY(`attachment_id`) REFERENCES comment_attachment(aId)     \
 \    )"
 
-insertTicketInfo :: forall m. (MonadIO m) => Connection -> TicketInfo -> m ()
+insertTicketInfo :: forall m. (MonadIO m, MonadCatch m) => Connection -> TicketInfo -> m ()
 insertTicketInfo conn TicketInfo{..} =
-    liftIO $ executeNamed conn "INSERT INTO ticket_info (tiId, tiRequesterId, tiAssigneeId, tiUrl, tiTags, tiStatus) \
+    checkError (InsertTicketInfoException tiId) $ executeNamed conn "INSERT INTO ticket_info (tiId, tiRequesterId, tiAssigneeId, tiUrl, tiTags, tiStatus) \
```

Comment:
This seems too heavy. The nice thing about functions is that you can compose them easily. 
Focus on making a single function do what it's supposed to do, not having a function that does everything for every case?
I'll be more specific after you try again.